firmware/library/L3_HAL/sddriver.hpp

```diff
@@ -0,0 +1,239 @@
+        kGetStatus  = 0x04 | 13,    // CMD13: get status register
+        kChgBlkLen  = 0x40 | 16,    // CMD16: change block length (only
+                                    // effective in SDSC cards; SDHC/SDXC
+                                    // cards are locked to 512-byte blocks)
+        kReadBlk    = 0x40 | 17,    // CMD17: read a single block of data
+        kReadBlks   = 0x40 | 18,    // CMD18: read many blocks of data until
+                                    // a "CMD12" frame is sent
+        kWriteBlk   = 0x40 | 24,    // CMD24: write a single block of data
+        kWriteBlks  = 0x40 | 25,    // CMD25: write many blocks of data until
+                                    // a "CMD12" frame is sent
+        kDelFrom    = 0x40 | 32,    // CMD32: set address of the start block
+                                    // for deletion
+        kDelTo      = 0x40 | 33,    // CMD33: set address of the end block for
+                                    // deletion
+        kDel        = 0x40 | 38,    // CMD38: begin deletion from the block
+                                    // range specified by the vector
+                                    // [DEL_FROM : DEL_TO]
+        kAcBegin    = 0x40 | 55,    // CMD55: signals the start of an
+                                    // application-specific command
+        kGetOcr     = 0x40 | 58,    // CMD58: request data from the
+                                    // operational conditions register
+        kAcInit     = 0x40 | 41     // ACMD41: application-specific version of
+                                    // CMD1 (must precede with CMD55)
+    };
+
+    // @function        SendCmd()
+    // @description     sends a command frame to the SD Card and sets the
+    //                  response
+    // @parameter       (SdCommand) sdc                 the command code to
+    //                                                  send as the first byte
+    //                  (uint32_t) arg                  the 4-byte argument
+    //                                                  for the command
+    //                  ~(uint8_t*) response_buffer     pointer to a buffer to
+    //                                                  house the response
+    //                                                  frame
+    //                  ~(uint32_t) delay               delay (ms) between
+    //                                                  asserting CS and
+    //                                                  sending a cmd frame
+    //                  ~(bool) keep_alive              determines whether to
+    //                                                  de-assert CS after
+    //                                                  receiving response
+    // @returns         on success: the number of bytes given in the response
+    //                  on failure: -1
+    virtual uint32_t SendCmd(
+        SdCommand sdc,
+        uint32_t arg,
```

Comment:
Spell words out fully, do not use short hand like arg, for arguments.

Reply:
I also prefer, parameters over arguments.

Reply:
Is each byte an argument (4 arguments total) or is the whole thing (1 argument)? If its 4 arguments, then a `struct` similar to :

SJ2_PACKED(struct) Parameters_t
{
     uint8_t parameter[4];
};